tests/freshness.rs

```diff
@@ -172,6 +172,13 @@ fn changing_features_is_ok() {
 [FINISHED] debug [unoptimized + debuginfo] target(s) in [..]
 "));
 
+    /* Targets should be cached from the first build
+       XXX Sadly these cannot be cached since the "symlink" step is
+           not separate from the "compile" step. Packages which link
+           to top level binaries eg (deps/foo-abc123 -> foo) are forced
+           to do an extra recompile here.
+    */
```

Comment:
If we modify the "symlink" step to also happen in the "fresh" unit of work I think we can solve this, right?